eval, read and minor tweaks

DaveWoodman opened the pull request:
Changes to clips.jj to return more when using (read)
Fix for eval (getVal fix).
Minor tweaks

Commit:
Misc fixes

Assert prints error message if template not found
Clear (clearAll) now clears away modules
Various unsafe cast tweaks

DaveWoodman committed:
Merge branch 'woolfel:main' into main

Commit:
Update for clearAll to use main module named in Constants.java

DaveWoodman committed:
Delete jline-3.19.0.jar

Commit:
Fix to STRING_LITERAL vs INDETIFIER parsing

Change to the parser specification so that IDENTIFIERs are returned
correctly. Previously an IDENTIFIER was treated as STRING_LITERAL and
had the first and last chars stripped since these were assumed to be
quotes. This caused a problem with (read) (e.g. (read)ing Fred returned
re).

Commit:
Merge branch 'main' of https://github.com/DaveWoodman/morendo.git into main

Commit:
Minor changes and eval fix

Changes to the general case for getValue - other minor tweaks

woolfel commented on src/main/org/jamocha/parser/clips/clips.jj:
```diff
@@ -299,7 +304,16 @@ Object basicExpr() :
+    if (retval != null) return retval;
+    else throw new ParseException();
+  }
+} 
```
I haven't tested it, but wouldn't this always return a token when the user presses a key?

DaveWoodman replied:
Only tokens that are not specifically handled are returned when the user enters a string and the return key, String literals  ( ) enclosed and valid identifiers are handled specifically. 

The previous version would assume that anything not known was a string and strip the first and last characters in the mistaken believe that they were quotes. Now the string will be returned as entered if not specifically handled.

Commit:
Fix ~nil and pretty print for literal constraints for ~nil

Commit:
Betterequality check for NIL_SYMBOL

Commit:
(insert$) CLIPS compatability

Now adds values to end of array if specified insertion point is greater
than its length

Commit:
Upgrade to JavaCC 7.0.10 - parse error handling improvement

Commit:
Completion of mighation to Javacc 7.0.10
